aten/src/ATen/native/mps/operations/Copy.mm

```diff
@@ -42,60 +42,91 @@ bool is_strided_contiguous(const at::Tensor& t) {
   return compute_strided_size(t) == t.numel();
 }
 
+static char* COPY_CAST_OP_TEMPLATE_TENSOR = R"METAL_COPY_CAST(
+kernel void copy_cast_kernel(uint tid              [[thread_position_in_grid]],
+                       const device {0} * input   [[buffer(0)]],
+                       device       {1} * output  [[buffer(1)]]) {{
+  output[tid] = ({1})input[tid];
+}}
+)METAL_COPY_CAST";
+
+static
+id<MTLLibrary> compileCopyCastOpsLibrary(id<MTLDevice> device,
```

Comment:
Looks good. But we should definitely refactor all of these repetitive compileLibrary/getPipeline state functions along with their duplicate cachings code in a common place like MPSKernel.mm later.